### ISSUE TITLE
Bidirectional Org Chart ( Two Layouts at a time )

### ISSUE THREAD

**Issue by bhavyakelp** [bumbeishvili/org-chart#419]
Hi

I am wanting to create a bidirectional org chart ( similar to a family tree), which has the root node at the center, its children below it and parents above it
If I further expand my parent nodes they open upwards and if I open child node they open downwards.

In the org chart example that is posted ([here](https://stackblitz.com/edit/web-platform-o5t1ha?file=README.md)),

- If the layout is bottom : It opens the chart in bottom up direction ( upwards )
<img width="895" alt="Screenshot 2024-07-16 at 4 20 08 PM" src="https://github.com/user-attachments/assets/e535824f-ec1f-4104-bc07-0628061e981f">

- and if the layout is top : It opens the chart in top down direction (downwards)

<img width="712" alt="Screenshot 2024-07-16 at 4 20 31 PM" src="https://github.com/user-attachments/assets/a50038dc-929e-44c1-a169-2c50d31018e0">

what I aim to do is have both of these in one single chart and create something similar to this:
 - with the yellow node being the root
 - purple being the children
 - green being the parent
 - and a expand (+)/ collapse(-) icon to toggle between expanded states

![image](https://github.com/user-attachments/assets/575c6793-cc91-423f-ae2c-593d1c306819)

How will be able to achieve this ? I tried multiple ways but am unable to attain this behaviour.......

**Comment by bumbeishvili** [comment]
Hi, it's not easily possible with this library, and not sure how to approach it at the moment

**Comment by bhavyakelp** [comment]
@bumbeishvili are there any algorithms / packages that can help me achieve this ?

**Comment by bumbeishvili** [comment]
No idea

**Comment by bumbeishvili** [comment]
I am guessing it will be along these lines

https://github.com/bumbeishvili/org-chart/blob/cb1f376656bc83360a4fea6872597fd427fcd775/src/d3-org-chart.js#L838-L848

**Comment by bhavyakelp** [comment]
That helps, @bumbeishvili!

In the meantime, I was browsing through some of your projects, and I find them truly amazing! 
But, I came across your [secureLogic visualization](https://davidb.dev/project/securelogic), which is very similar to what I need.

<img width="1429" alt="Screenshot 2024-07-18 at 10 22 28 PM" src="https://github.com/user-attachments/assets/ed82437e-f8af-49f7-8885-17dfcc8933df">

If you see this chart is similar to an org chart but works in two directions (LR and RL)
So if I reciprocate this chart, I can then make it TB and BT, add my custom nodes and voila!

Is that a good idea and is it possible? I would like your view and input on this ......

**Comment by bumbeishvili** [comment]
Yes, it's similar, as I remember I did it from scratch, instead of using this library, I don't exactly remember how I did it, but If I were to do it again, I'd use the same Flextree library, I'd probably just change the direction of some branches from positive to negative coordinates

https://observablehq.com/@bumbeishvili/horizontal-collapsible-d3-flextree
